src/core/Tools.cpp

```diff
@@ -56,294 +57,161 @@
+    QString base64 = QString::fromLatin1(ba.constData(), ba.size());
 
-// Mac OS X
-#ifdef HAVE_PT_DENY_ATTACH
-        success = success && (ptrace(PT_DENY_ATTACH, 0, 0, 0) == 0);
-#endif
+    return regexp.exactMatch(base64);
+}
 
-#ifdef Q_OS_WIN
-        success = success && createWindowsDACL();
-#endif
+void sleep(int ms)
+{
+    Q_ASSERT(ms >= 0);
 
-        if (!success) {
-            qWarning("Unable to disable core dumps.");
-        }
+    if (ms == 0) {
+        return;
     }
 
-    void setupSearchPaths()
-    {
 #ifdef Q_OS_WIN
-        // Make sure Windows doesn't load DLLs from the current working directory
-        SetDllDirectoryA("");
-        SetSearchPathMode(BASE_SEARCH_PATH_ENABLE_SAFE_SEARCHMODE);
+    Sleep(uint(ms));
+#else
+    timespec ts;
+    ts.tv_sec = ms/1000;
+    ts.tv_nsec = (ms%1000)*1000*1000;
+    nanosleep(&ts, nullptr);
 #endif
-    }
-
-    //
-    // This function grants the user associated with the process token minimal access rights and
-    // denies everything else on Windows. This includes PROCESS_QUERY_INFORMATION and
-    // PROCESS_VM_READ access rights that are required for MiniDumpWriteDump() or ReadProcessMemory().
-    // We do this using a discretionary access control list (DACL). Effectively this prevents
-    // crash dumps and disallows other processes from accessing our memory. This works as long
-    // as you do not have admin privileges, since then you are able to grant yourself the
-    // SeDebugPrivilege or SeTakeOwnershipPrivilege and circumvent the DACL.
-    //
-    bool createWindowsDACL()
-    {
-        bool bSuccess = false;
-
-#ifdef Q_OS_WIN
-        // Process token and user
-        HANDLE hToken = nullptr;
-        PTOKEN_USER pTokenUser = nullptr;
-        DWORD cbBufferSize = 0;
-
-        // Access control list
-        PACL pACL = nullptr;
-        DWORD cbACL = 0;
-
-        // Open the access token associated with the calling process
-        if (!OpenProcessToken(GetCurrentProcess(), TOKEN_QUERY, &hToken)) {
-            goto Cleanup;
-        }
-
-        // Retrieve the token information in a TOKEN_USER structure
-        GetTokenInformation(hToken, TokenUser, nullptr, 0, &cbBufferSize);
-
-        pTokenUser = static_cast<PTOKEN_USER>(HeapAlloc(GetProcessHeap(), 0, cbBufferSize));
-        if (pTokenUser == nullptr) {
-            goto Cleanup;
-        }
-
-        if (!GetTokenInformation(hToken, TokenUser, pTokenUser, cbBufferSize, &cbBufferSize)) {
-            goto Cleanup;
-        }
-
-        if (!IsValidSid(pTokenUser->User.Sid)) {
-            goto Cleanup;
-        }
+}
 
-        // Calculate the amount of memory that must be allocated for the DACL
-        cbACL = sizeof(ACL) + sizeof(ACCESS_ALLOWED_ACE) + GetLengthSid(pTokenUser->User.Sid);
-
-        // Create and initialize an ACL
-        pACL = static_cast<PACL>(HeapAlloc(GetProcessHeap(), 0, cbACL));
-        if (pACL == nullptr) {
-            goto Cleanup;
-        }
+void wait(int ms)
```

Comment:
Can we use AsyncTask for this instead of a new solution?

Reply:
I think wait is only used in tests. I removed one or two uses, but didn't bother with the rest. I don't even know if it's really needed. It is definitely not new, though.

Reply:
It's only used in a few places where I can't find a working alternative, so I don't think I should waste time on this.